wazuh/wazuh-splunk
# 1174 [RBAC] Implement auth context login

closes #1174

This PR provides a second login method for the users, by using an [authorization context](https://documentation.wazuh.com/current/user-manual/api/rbac/auth-context.html).

The user's token provided by the Wazuh API is managed by the Python backend on the Splunk's App. By default, the [auth context login](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.security_controller.run_as_login) is prefered, and if it fails, a basic login is performed instead.

The authorization context provided to the Wazuh API is the internal Splunk's username, as follows:

```json
{
    "username": "admin"
}
```

## Testing
Proper testing must be done with the integration of the Security section.

As for now:

1. (Optional) Set the log level to `debug`, either on `Settings < Configuration` or directly on the `config.conf` file. Indexer restart is required.
2. Open and watch the log file: `docker-compose exec -u root indexer tail -f /opt/splunk/var/log/splunk/SplunkAppForWazuh.log`
3. Navigate through the app and check for errors on the logs.

The step 1 can be ignored as the last commit mistakenly included the debug log level. This is to be fixed later.
